src/programmatic_policy_learning/approaches/llm_grid_search_approach.py

```diff
@@ -0,0 +1,89 @@
+"""LLM + 1D grid search.
+
+This program ties together:
+  - synthesize_llm_parametric_policy (single prompt → parametric policy)
+  - grid_search_param (generic 1D grid search)
+
+And it returns the tuned policy instance (recreated with the best param), the
+best parameter value, and the average return at that value.
+"""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Dict, Optional, Tuple, cast
+
+from gymnasium.spaces import Box
+from prpl_llm_utils.models import PretrainedLargeModel
+
+from programmatic_policy_learning.approaches.grid_search_approach import (
+    grid_search_param,
+)
+from programmatic_policy_learning.approaches.llm_single_search_approach import (
+    LLMGeneratedParametricPolicy,
+    synthesize_llm_parametric_policy,
+)
+from programmatic_policy_learning.approaches.structs import ParametricPolicyBase
+
+
+def synthesize_and_grid_search(
+    env_factory: Callable[[], Any],
+    environment_description: str,
+    action_space: Box,
+    llm: PretrainedLargeModel,
+    example_observation: Any,
+    param_name: str = "kp",
```

Comment:
let's make this a required argument because "kp" is only a sensible default for pendulum